envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsAllowedClientTest.kt

```diff
@@ -0,0 +1,282 @@
+        )
+
+        // then
+        assertThat(echoResponse).isForbidden()
+        assertThat(echoEnvoy.container).hasOneAccessDenialWithActionBlock(
+            protocol = "http",
+            path = "/block-unlisted-clients",
+            method = "GET",
+            clientName = "echo3",
+            trustedClient = false,
+            clientAllowedToAllEndpoints = true,
+            clientIp = echoEnvoy.container.gatewayIp()
+        )
+    }
+
+    @Test
+    fun `echo2 should allow special client with client identity header over https and log request when unlistedEndpointsPolicy is log`() {
+        // when
+        val echo2Response = echo2Envoy.ingressOperations.callLocalServiceInsecure(
+            endpoint = "/log-unlisted-endpoint",
+            headers = Headers.of(mapOf("x-service-name" to "echo3")),
+            useTls = true
+        )
+
+        // then
+        assertThat(echo2Response).isOk().isFrom(service)
+        assertThat(echo2Envoy.container.ingressSslRequests).isOne()
+        assertThat(echo2Envoy.container).hasOneAccessDenialWithActionLog(
+            protocol = "https",
+            path = "/log-unlisted-endpoint",
+            method = "GET",
+            clientName = "echo3 (not trusted)",
+            trustedClient = false,
+            clientAllowedToAllEndpoints = true,
+            clientIp = echo2Envoy.container.gatewayIp()
+        )
+    }
+
+    @Test
+    fun `echo should block special client with client identity header over https and log request when unlistedEndpointsPolicy is blockAndLog`() {
+        // when
+        val echoResponse = echoEnvoy.ingressOperations.callLocalServiceInsecure(
+            endpoint = "/block-and-log-unlisted-endpoint",
+            headers = Headers.of(mapOf("x-service-name" to "echo3")),
+            useTls = true
+        )
+
+        // then
+        assertThat(echoResponse).isForbidden()
+        assertThat(echoEnvoy.container.ingressSslRequests).isOne()
+        assertThat(echoEnvoy.container).hasOneAccessDenialWithActionBlock(
+            protocol = "https",
+            path = "/block-and-log-unlisted-endpoint",
+            method = "GET",
+            clientName = "echo3 (not trusted)",
+            trustedClient = false,
+            clientAllowedToAllEndpoints = true,
+            clientIp = echoEnvoy.container.gatewayIp()
+        )
+    }
+
+    @BeforeEach
+    fun startRecordingRBACLogs() {
+        echoEnvoy.recordRBACLogs()
+        echo2Envoy.recordRBACLogs()
+    }
+
+    @AfterEach
+    fun cleanupTest() {
+        echoEnvoy.container.admin().resetCounters()
+        echoEnvoy.container.logRecorder.stopRecording()
+        echo2Envoy.container.admin().resetCounters()
+        echo2Envoy.container.logRecorder.stopRecording()
+    }
+
+    private val EnvoyContainer.ingressSslRequests: Int?
+        get() = this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
```

Comment:
Why a property with an overridden getter instead of a function? Also, we've decided to use `TLS` instead of `SSL`.
```suggestion
    private fun EnvoyContainer.ingressTlsRequests() = 
        this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
```